e2e/src/test/java/no/sysco/testing/kafka/e2e/ContainersIT.java

```diff
@@ -0,0 +1,194 @@
+package no.sysco.testing.kafka.e2e;
+
+import io.restassured.RestAssured;
+import io.restassured.http.ContentType;
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import no.sysco.testing.kafka.e2e.representation.MessageJsonRepresentation;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.testcontainers.containers.BindMode;
+import org.testcontainers.containers.Container;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.KafkaContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
+
+import static org.awaitility.Awaitility.await;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+public class ContainersIT {
+
+    public static final String LOCAL_HOST = "http://localhost";
+    public static final String TOPIC = "events-message-v1";
+    public static final String CONFLUENT_PLATFORM_VERSION = "5.1.1";
+
+    // kafka env
+    public static KafkaContainer kafka;
+    public static String KAFKA_BROKER_INSIDE_DOCKER_ENV;
+    public static GenericContainer schemaRegistry;
+    public static String SCHEMA_REGISTRY_INSIDE_DOCKER_ENV;
+
+    // app env
+    public static GenericContainer jsonServer;
+    public static String JSON_SERVER_INSIDE_DOCKER_ENV;
+    public static String JSON_SERVER_EXPOSED;
+
+    public static GenericContainer httpProducer;
+    public static String HTTP_PRODUCER_EXPOSED;
+
+    public static GenericContainer httpMaterializer;
+
+    @BeforeClass
+    public static void start() {
```

Comment:
nit: I'd recommend to add all the docker container config here, instead of partitioned methods. Then you expose the configs on the same place and avoid using static variables.

Reply:
Do you mean put all code from partitioned methods under `start()` ?

Reply:
yeap, explaining/organazing with comments instead of multiple methods.

Reply:
yeah. I also though about creation abstract class, and than extract all `@BeforeClass` actions to that abstract class.